octomap/include/octomap/OcTreeBaseImpl.hxx

```diff
@@ -705,9 +705,13 @@ namespace octomap {
       // TODO delete check depth, what happens to inner nodes with children?
       this->deleteNodeChild(node, pos);
 
-      if (!nodeHasChildren(node))
+      if (!nodeHasChildren(node)){
```

Comment:
Is there a reason not to do this cleanup in deleteNodeChild() already?
It semantically belongs to the deletion of the (last) child, and doing it outside may lead to having to add the same routine to other contexts from where deleteNodeChild is called.

Reply:
Good point! The only reason I can think of is the (probably) less efficient call of "prune(...)", which does the cleanup once after deleting all 8 child nodes (instead of checking for every single call of deleteNodeChild if it's empty already).

Reply:
True, when you know you are deleting multiple children, you may not want to check after each deletion (which means iterating the 8 child pointers).

So I would suggest adding a parameter to `deleteNodeChild()` whether to cleanup, which in my opinion should have a default of `true`.